README and BREAKING_CHANGES log changes (#43191)
Azure/azure-sdk-for-net: sdk/identity/Azure.Identity/BREAKING_CHANGES.md

# Breaking Changes

## 1.11.0

### Behavioral change to `DefaultAzureCredential` in IMDS managed identity scenarios

> As of `Azure.Identity` 1.11.0, the `DefaultAzureCredential` makes a couple minor behavioral changes to request timeout and retry behavior in environments where IMDS managed identity is used.
> The changes are as follows:
> - The first request made to IMDS managed identity will be made with a 1 second timeout, as it did previously, but without the "Metadata" header in order to speed up the validation that the endpoint is available. This is guaranteed to fail with a 400 error.
>     - If the request times out, indicating that the IMDS endpoint is not available, no retries will be made. This is a change from the previous behavior, where the request would be retried up to 3 times, with exponential backoff.
>     - If the request returns a 400 error, indicating that the IMDS endpoint is available, the request will be retried up to 4 times, with exponential backoff, to allow for transient failures.

Reviewer: ```suggestion
As of `Azure.Identity` 1.11.0, the `DefaultAzureCredential` makes a couple minor behavioral changes to request timeout and retry behavior in environments where IMDS managed identity is used. The changes are as follows:
- The first request made to IMDS managed identity will be made with a 1-second timeout, as it did previously, but without the "Metadata" header to expedite validating whether the endpoint is available. This is guaranteed to fail with a 400 error.
    - If the request times out, indicating that the IMDS endpoint isn't available, no retries will be made. This is a change from the previous behavior, where the request was retried up to 3 times, with exponential backoff.
    - If the request returns a 400 error, indicating that the IMDS endpoint is available, the request will be retried up to 4 times, with exponential backoff, to allow for transient failures.
```
